### PR TITLE
LIME-16: Add Sign In Page

### LINKED ISSUE #16: Sign In Page
- Create a form and page according to the design. Exclude the "Continue with Google" and "Continue with Facebook" buttons.
- Implement the sign-in action flow. User should be added to Redux store upon successful login
- Add validation. Should be shared
- "Create and account" link should redirect to sign in page
 
**SPECS from QA Team**

**1.** Registered with email user can login using Login form

**2.** Form contains Email and Password fields, both mandatory

**Field  “Email”:**
- Placeholder should be as designed
- Should have minlength 6 characters and maxlength 320
- Should contain local part, @, domain part. 
**Local part:**
- Should contain from 1 to 64 characters.
- Uppercase and lowercase Latin letters (A-Z, a-z), digits (0-9), middle-placed non-consecutive dot (.), middle-placed hyphen (-), and underscore (_) are allowed, but not required
**Domain part:**
- Should contain from 4 to 254 characters.
- Uppercase and lowercase Latin letters (A-Z, a-z), digits 0-9, middle-placed hyphen (-) are allowed, but not required.
- Middle-placed non-consecutive dots (.) are allowed. One middle-placed dot is required.
Spaces before and after Email string should be cropped 

**Field  “Password”**
- Placeholder is 6 bullets as designed
- Entered text is displayed masked as bullets for security. User can review unmasked password after clicking on “eye” icon 
- Should have minlength 6 characters and maxlength 12
- Latin letters (A-Z, a-z), digits (0-9) and special characters are allowed
- Spaces before and after Password string should be cropped

**3.** Error messages:
- displayed under field with error value 
- for empty fields - Please fill out this field
- for invalid email - Please enter a valid email
- for invalid password length - Password must have from 6 to 12 characters
- for incorrect email or password - Incorrect email or password. Please try again. (displayed only under password field) In case of this error, form is displayed with saved values in the fields

**4.** “Log in” button:

- Should be inactive with empty mandatory fields
- Should be inactive after negative validation of the length and format of email or password 
- Should be inactive after negative validation of the correctness of email or password.

**5.** User can login using social media (Google, Facebook).

Image
[Design](https://www.figma.com/file/RRE9g9LIo3a1AK89HBlN6U/BSA_Lime?type=design&node-id=0-1&mode=design)

### PR DESCRIPTION
**Pull request by stpalamar**
Everything is ready, ~~but I need password input type. So I will commit this change when new input component will be ready~~

**EDITED: Some updates in comments!**

### CODE REVIEW

**Comment by SofiiaTrokhymchuk** [comment]
Updated Loader component with new logo is already merged, so you can use it

File: frontend/src/bundles/auth/pages/auth.tsx [hunk 1]
```diff
@@ -41,11 +63,26 @@ const Auth: React.FC = () => {
         return null;
     };
 
+    const classes = {
+        base: 'flex flex-col flex-1 bg-lm-black-200 mx-[1rem] my-[1.125rem] rounded-[2.75rem] lg:flex-none lg:w-[44rem] lg:px-[11.25rem]',
+        loader: 'justify-center items-center',
```
Comment (SofiiaTrokhymchuk) [inline frontend/src/bundles/auth/pages/auth.tsx]: You can use `<Loader isOverflow />` to display it centered inside of the block. The styles of the overflow Loader use absolute position, so your parent needs to be relative. It is absolute inside parent and is already centered. You can try this

Comment (stpalamar) [reply frontend/src/bundles/auth/pages/auth.tsx]: Thanks for the advice! I will change it

**Comment by stpalamar** [comment]
I removed error message in form. Notification toast will be automatically displayed when #25 will be merged

**Comment by stpalamar** [comment]
Moved loader inside in button. Inputs and button are disabled while request is pending. I guess everything is done
image

**Review by dimapopovych — APPROVED** [review]

**Review by anton-otroshchenko — APPROVED** [review]

File: frontend/src/bundles/auth/store/actions.ts [hunk 2]
```diff
@@ -2,20 +2,30 @@ import { createAsyncThunk } from '@reduxjs/toolkit';
...
 
-export { signUp };
+const signIn = createAsyncThunk<
+    UserAuthResponseDto,
+    UserAuthRequestDto,
+    AsyncThunkConfig
+>(`${sliceName}/sign-in`, (loginPayload, { extra }) => {
+    const { authApi } = extra;
+
+    return authApi.signIn(loginPayload);
```
Comment (dimapopovych) [inline frontend/src/bundles/auth/store/actions.ts]: Merge dev and resolve conflicts, also token should be set to local storage here

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**